File: exercicio04/java/Figura3.java
package exercicio04.java;

public class Figura3 {
    public double coordX;
    public double coordY;

    public Figura3(double x, double y) {
        this.coordX = x;
        this.coordY = y;
    }

    public void desenhar() {
        System.out.println("Desenho das coordenadas (" + this.coordX + ", " + this.coordY + ")");
    }

    public void pintar() {
        System.out.println("Pintando os pontos das coordenadas (" + this.coordX + ", " + this.coordY + ")");
    }

    public void mover() {
        this.coordX += 2;
        this.coordY += 2;
        System.out.println("Figura movida\nNovas coordenadas:");
        this.imprimi();
    }

    public void quadrante() {
        if (this.coordX > 0 && this.coordY > 0) {
            System.out.println("A figura está no primeiro quadrante");
        } else if (this.coordX < 0 && this.coordY > 0) {
            System.out.println("A figura está no segundo quadrante");
        } else if (this.coordX < 0 && this.coordY < 0) {
            System.out.println("A figura está no terceiro quadrante");
        } else {
            System.out.println("A figura está no quarto quadrante");
        }
    }

    public void imprimi() {
        System.out.printf("Coordenadas_da_Figura : (%f, %f)%n ", this.coordX, this.coordY);
    }
}

File: exercicio04/python/figura3.py
class Figura3:

    # construtor
    def __init__(self, x, y):
        self.cordX = x
        self.cordY = y

    def desenhar(self):
        print(f"Desenhando cordenadas({self.cordX}, {self.cordY}) ")

    def pintar(self):
        print(f"Pintando f_01 nas cordenadas({self.cordX}, {self.cordY})  ")

    # metodo_mover
    def mover(self):
        self.cordX += 2
        self.cordY += 2
        print("Figura movida\nNovas coordenadas:")
        self.get_coodenadas()

    # metodo_quadrante
    def quadrante(self):
        if self.cordX > 0 and self.cordY > 0:
            print('Figura no Quadrante 1!')
        elif self.cordX < 0 and self.cordY > 0:
            print('Figura no Quadrante 2!')
        elif self.cordX < 0 and self.cordY < 0:
            print('Figura no Quadrante 3!')
        elif self.cordX == 0 and self.cordY == 0:
            print('A Figura esta sobre o Eixo, a origem do Plano Cartesiano !!!')
        else:
            print('Figura no Quadrante 4!')

    def get_coodenadas(self):
        print(f"coordenadas(X, Y): ({self.cordX}, {self.cordY})")
